### PR TITLE
Fix some bugs with mavlink2 support

### LINKED ISSUE #1388: Linux binary crashing with PX4 SITL
Trying to run the Linux Binary with PX4 SITL setup and it crashes before starting. It works fine without the SITL setup though. 
Here'e the error log. 
linuxerror

### PR DESCRIPTION
**Pull request by clovett**
This pull request fixes some bugs in mavlinkcom support for mavlink 2.  Specifically it was not handling the new MAVLINK_MSG_ID_AUTOPILOT_VERSION message which tells all MavLinkConnections what versionh we are using (including those that are joined).  This might also fix [GitHub issue 1388](https://github.com/Microsoft/AirSim/issues/1388) and [1516](https://github.com/Microsoft/AirSim/issues/1516).

### CODE REVIEW

File: MavLinkCom/src/impl/MavLinkNodeImpl.cpp [hunk 1]
```diff
@@ -58,7 +58,15 @@ void MavLinkNodeImpl::sendHeartbeat()
...
         heartbeat.system_status = 0; // ignored by PX4
-        sendMessage(heartbeat);
+        try 
+        {
+            sendMessage(heartbeat);
+        }
+        catch (std::exception& e)
+        {
+            // ignore any failures here because we are running in our own thread here.
+            Utils::log(Utils::stringf("Caught and ignoring exception sending heartbeat: %s", e.what()));
```
Comment (sytelus) [inline MavLinkCom/src/impl/MavLinkNodeImpl.cpp]: This is bit worry some. Why would exception occur and is there is no way to transfer this information somehow up the chain? What would be the worse case scenario here? May be exception occurs and user gets confused that nothing is happening?

File: MavLinkCom/src/impl/MavLinkVehicleImpl.cpp [hunk 2]
```diff
@@ -148,7 +148,8 @@ AsyncResult<bool> MavLinkVehicleImpl::allowFlightControlOverUsb()
 
 void MavLinkVehicleImpl::handleMessage(std::shared_ptr<MavLinkConnection> connection, const MavLinkMessage& msg)
 {
-    unused(connection);
+    MavLinkNodeImpl::handleMessage(connection, msg);
```
Comment (sytelus) [inline MavLinkCom/src/impl/MavLinkVehicleImpl.cpp]: Any idea why this change?

File: MavLinkCom/src/MavLinkMessages.cpp [hunk 3]
```diff
@@ -5101,6 +5101,36 @@ std::string MavLinkDebug::toJSon() {
  return ss.str();
 }
 
+int MavLinkProtocolVersion::pack(char* buffer) const {
```
Comment (sytelus) [inline MavLinkCom/src/MavLinkMessages.cpp]: Would all these continue to work with MavLink 1.0 as well?

**Review by sytelus — COMMENTED** [review]
Overall looks good and if you are happy we can merge it.

File: MavLinkCom/src/impl/MavLinkNodeImpl.cpp (hunk 1 above)
Comment (clovett) [reply MavLinkCom/src/impl/MavLinkNodeImpl.cpp]: I agree it is not ideal, but this function is running in an independent thread, and there is no other thread that we can transfer this to.

File: MavLinkCom/src/impl/MavLinkVehicleImpl.cpp (hunk 2 above)
Comment (clovett) [reply MavLinkCom/src/impl/MavLinkVehicleImpl.cpp]: MavLinkNodeImpl needs the handleMessage also.

File: MavLinkCom/src/MavLinkMessages.cpp (hunk 3 above)
Comment (clovett) [reply MavLinkCom/src/MavLinkMessages.cpp]: yes.

**Comment by Vinay545** [comment]
Are the linux binaries compiled with this patch. Because I am facing this issue with the existing binary releases. If not, when the new binaries that are compiled with this patch will be released.

**Pull request merged**